fluss-server/src/test/java/com/alibaba/fluss/server/replica/ReplicaManagerTest.java

```diff
@@ -1163,6 +1166,51 @@ void testStopReplica() throws Exception {
         replicaManager.getReplicaOrException(tb);
     }
 
+    @Test
+    void testKvDataVisibility() throws Exception {
+        // The CDC log is only visible after the KV has been flushed to RocksDB. In other words,
+        // when we can read the CDC log, the associated kv record must have been
+        // inserted/updated/deleted in RocksDB. The reason for ensuring this visibility is that we
+        // first buffer the data in memory before flushing it to RocksDB. Thus, we need to guarantee
+        // visibility.
+        TableBucket tb = new TableBucket(DATA1_TABLE_ID_PK, 1);
+        makeKvTableAsLeader(DATA1_TABLE_ID_PK, DATA1_TABLE_PATH_PK, tb.getBucket());
+        Replica replica = replicaManager.getReplicaOrException(tb);
+
+        // retry send kv records to kv store, if the highWatermark increased, the kv record must be
+        // visible in rocksdb.
+        int round = 1000;
+        CompactedKeyEncoder keyEncoder = new CompactedKeyEncoder(DATA1_ROW_TYPE, new int[] {0});
+        CompletableFuture<List<PutKvResultForBucket>> future;
+        for (int i = 0; i < round; i++) {
+            future = new CompletableFuture<>();
+            Object[] key = {i};
+            Object[] value = {i, "a"};
+            byte[] keyBytes = keyEncoder.encodeKey(row(key));
+            byte[] valueBytes =
+                    ValueEncoder.encodeValue(
+                            DEFAULT_SCHEMA_ID, compactedRow(DATA1_SCHEMA_PK.getRowType(), value));
+            List<byte[]> lookups = replica.lookups(Collections.singletonList(keyBytes));
+            assertThat(lookups.size()).isEqualTo(1);
+            assertThat(lookups.get(0)).isNull();
+            replicaManager.putRecordsToKv(
+                    20000,
+                    1,
```

Comment:
this test can reproduce the problem, because it only has one replia and ack=1, so after executing `replicaManager.putRecordsToKv`, the watermark has been advanced, and kv data has been flushed, thus the following test is always success without any changes to the source code.

Reply:
Yes, I switched to a multi-thread way to validate this issue. In the new test, if `highWatermarkUpdateTimestamp` is less than `lastTimestampForNullValue` even once, it indicates that the CDC log visibility is earlier than the KV. Specifically, `highWatermarkUpdateTimestamp` is later than the actual `highWatermarkUpdate` time, and `lastTimestampForNullValue` is earlier than the collected time, which is intended to eliminate influence of collecting. Pr ready.

Reply:
In my local env, if you revert the change, the test will always failed.